RAK4631 ESM Migration

cod3doomy opened the pull request:
Changes to migrate sensor code to the ESM.

Added a separate GPS init sequence for the RAK that scans I2C and Serial1 on the various sockets of the various base boards to find the RAK12500. (and soon the RAK12501)

Removed the GPS specific envs from platformio.ini and enabled GPS for all envs.

Verified working with RAK12500 on RAK19007 sockets A and D, as well as RAK19003.

ripplebiz commented on src/helpers/sensors/EnvironmentSensorManager.cpp:
```diff
@@ -296,8 +312,87 @@ void EnvironmentSensorManager::initBasicGPS() {
+  //set initial waking state
+  pinMode(ioPin,OUTPUT);
+  digitalWrite(ioPin,LOW);
+  delay(1000);
```
do these delays have to be so long?

ripplebiz commented on variants/rak4631/platformio.ini:
```diff
@@ -5,21 +5,27 @@ board = wiscore_rak4631
 board_check = true
 build_flags = ${nrf52_base.build_flags}
   -I variants/rak4631
-  -D RAK_4631
```
There are still other modules that rely on this RAK_4631 macro.
Found in: ArduinoSerialInterface

446564 commented:
confirmed working on 19007 with 1901 module (SHTC3). still have to manually add the lib dep and build flag to enable it, but I suppose that is a future change to automate all that.

cod3doomy commented:
Cleaned up code and added fixes/features that were requested. The 1901 should now be usable, along with the 1902 and 16000, although I don't have those three sensors to test myself.
